connect-inject/container_init.go

```diff
@@ -78,6 +87,7 @@ func (h *Handler) containerInit(pod corev1.Pod, k8sNamespace string) (corev1.Con
 		NamespaceMirroringEnabled: h.EnableK8SNSMirroring,
 		ConsulCACert:              h.ConsulCACert,
 		EnableTransparentProxy:    tproxyEnabled,
+		EnvoyUID:                  strconv.Itoa(envoyUserAndGroupID),
```

Comment:
I think we can pass it in as an `int` since the template will render it corrctly.